src/vs/workbench/browser/parts/editor/noEditorTabsControl.ts

```diff
@@ -19,10 +20,27 @@ export class NoEditorTabsControl extends EditorTabsControl {
 	}
 
 	openEditor(editor: EditorInput): boolean {
-		return false;
+		return this.handleOpenedEditors();
 	}
 
 	openEditors(editors: EditorInput[]): boolean {
+		return this.handleOpenedEditors();
+	}
+
+	private handleOpenedEditors(): boolean {
+		const didChange = this.activeEditorChanged();
+		this.activeEditor = this.tabsModel.activeEditor;
+		return didChange;
+	}
+
+	private activeEditorChanged(): boolean {
```

Comment:
@benibenj can we try to reuse this code across all controls? I think we have it 3 times now:

<img width="692" alt="image" src="https://github.com/microsoft/vscode/assets/900690/edda203a-296d-4022-9dcc-ce86b9ba9e5f">

Reply:
Yes, have a note to do this as debt